app/src/main/java/com/example/sharingang/MapFragment.kt

```diff
@@ -45,11 +50,12 @@ class MapFragment : Fragment(), OnMapReadyCallback {
                 moveCameraToLastLocation()
                 hasCameraMovedOnce = true
             }
-            moveLastLocationMarker()
         }
     }
-    private var lastLocationMarker: Marker? = null
+
     private var map: GoogleMap? = null
+    private var markerManager: MarkerManager? = null
+    private var clusterManager: ClusterManager<MapItem>? = null
```

Comment:
Can we make those lateinit vars? To avoid having unnecessary null checks

Reply:
I really wanted to do that too but I think it's not possible because it requires the map and it's not initialized in `onCreateView`, it's done in a coroutine because we have to do it asynchronously

Reply:
A way to avoid all these null checks would be to pass around the `map` object to the functions that need it but I don't know if it's better. What do you think ?

Reply:
You're right, then I guess it's fine the way it is